src/index_runner/indexers/from_sdk.py

```diff
@@ -0,0 +1,151 @@
+"""
+Import 
+"""
+import uuid
+import json
+import docker
+import requests
+import tempfile
+from ..utils.config import get_config
+
+_CONFIG = get_config()
+_DOCKER = docker.from_env()
+_SCRATCH = _CONFIG['scratch']
+_NAMESPACE = "WS"
+
+
+def _get_docker_image_name(module_name, module_version=None):
+    """Query the Catalog to get the docker image name for the indexer application"""
+    catalog_service_url = _CONFIG['kbase-endpoint'] + '/catalog'
+    params = {
+        "method": "Catalog.get_module_version",
+        "version": "1.1",
+        "id": 'id',
+        "params": [{
+            "module_name": module_name
+        }]
+    }
+    if module_version != None:
+        params['params'][0]['version'] = module_version
+
+    resp = requests.post(catalog_service_url, json.dumps(params))
+    try:
+        json_resp = resp.json()
+    except Exception:
+        raise resp.text
+    result = json_resp['result'][0]
+    return result["docker_img_name"]
+
+
+def _verify_and_format_output(data_path, workspace_id, object_id):
+    """make sure the sdk indexers follow conventions, and stream the if necessary."""
+    def check_datatypes(d):
+        """verify that the outputs of the sdk indexer follows these rather strict conventions"""
+        if isinstance(d, dict):
+            for key, val in d.items():
+                # not sure if we want to do recursive here or not (not for now)
+                assert isinstance(key, str), "Keys returned from indexer must be strings"
+                assert isinstance(val, str) or isinstance(val, int) or isinstance(val, float) or val is None or isinstance(val, bool), \
+                       "Values returned from indexer must be strings, integers, floats or Nonetype"
+        if isinstance(d, list):
+            for val in d: check_datatypes(val)
+
+    def format_data(d):
+        sub_type = d.get('sub_type', None)
+        sub_id = d.get('sub_id', None)
+        if d.get('sub_type') and d.get('sub_id'):
+            es_id = f"{_NAMESPACE}::{workspace_id}:{object_id}::{d.get('sub_type')}::{d.get('sub_id')}"
+        else:
+            es_id = f"{_NAMESPACE}::{workspace_id}:{object_id}"
+        return {
+            "index": index_name + ":" + indexer_version,
```

Comment:
Where does "index_name" and "indexer_version" come from? I'm not seeing these vars defined